### PR TITLE
Adds Tagging module

### PR DESCRIPTION
**Pull request by vishalzambre**
PR for #7343 
Added `acts_as_taggable` to spree.
Integrated with product for now.

### CODE REVIEW

**Comment by zambrevishal** [comment]
@damianlegawiec  Can I resolve these comments for double-quoted for strings because in current code most of places have mixed format and as per Rubocop don't use double quote unless you are not interpolating string.

Please let me know.

**Comment by vishalzambre** [comment]
@damianlegawiec Can I resolve these comments for double-quoted for strings because in current code most of places have mixed format and as per Rubocop don't use double quote unless you are not interpolating string.

**Comment by damianlegawiec** [comment]
@vishalzambre - great work - my only concern is that you've added a migration for tags without the `spree` prefix

**Comment by vishalzambre** [comment]
@damianlegawiec Its because of I used acts_as_taggable gem and they have table names as I added in migration, but for user purpose I added Spree::Tag model which is inherited from gem Tag model.

I think It will not affect, If we need `spree` prefix then I need to override models from gem, Let me know, If it requires I'll do that.

**Comment by zambrevishal** [comment]
@damianlegawiec  I think it will not affect anything to have migrations without prefix ```spree```.

**Comment by vishalzambre** [comment]
@damianlegawiec  I think it will not affect anything to have migrations without prefix ```spree```.

**Comment by damianlegawiec** [comment]
@vishalzambr this will result in a lot of trouble for anybody who has already `acts_as_taggable` in their projects, that's why the prefix is a must IMO

@priyank-gupta your thoughts?

**Comment by priyank-gupta** [comment]
@damianlegawiec Agreed. This also helps in scoping migrations if already present with same name and easily identifiable as which gem/extension added that migration.

**Comment by vishalzambre** [comment]
@damianlegawiec  @priyank-gupta  I'll modify migrations and will add them under spree namespce.

**Comment by vishalzambre** [comment]
@damianlegawiec @priyank-gupta  Added `spree` prefix to migrations

**Comment by damianlegawiec** [comment]
Nice work @vishalzambre 👍

**Comment by vishalzambre** [comment]
@damianlegawiec @priyank-gupta   Hey, I'd be happy to solve bug or add features in spree if you have anything in bucket please assign me.

**Comment by tommiller1** [comment]
NOTE: Migration 20160511071954_acts_as_taggable_on_migration.rb from spree has been skipped. Migration with the same name already exists.
NOTE: Migration 20160511072249_change_collation_for_tag_names.rb from spree has been skipped.
Migration with the same name already exists.

Please change the file names too.

**Comment by vishalzambre** [comment]
@tommiller1 I checked in spree core db/migrations, I'm not getting any files having same name, and If that was the case then spec should be failed, Its passing spec

**Comment by tommiller1** [comment]
It clashes for people who already have acts_as_taggable installed.

**Comment by vishalzambre** [comment]
@tommiller1 Right, make sense, will raise new PR with changing migration names.

File: core/db/migrate/20160511072249_change_collation_for_tag_names.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This migration is added to circumvent issue #623 and have special characters
+# work properly
+class ChangeCollationForTagNames < ActiveRecord::Migration
+  def up
+    if ActsAsTaggableOn::Utils.using_mysql?
+      execute("ALTER TABLE spree_tags MODIFY name varchar(255) CHARACTER SET utf8 COLLATE utf8_bin;")
```
Comment (krzysiek1507) [inline core/db/migrate/20160511072249_change_collation_for_tag_names.rb]: This SQL causes DB fail on MySQL https://travis-ci.org/spree/spree_auth_devise/jobs/133343113

Comment (vishalzambre) [inline core/db/migrate/20160511072249_change_collation_for_tag_names.rb]: @krzysiek1507  If you checked log from travis-ci its executing ```execute("ALTER TABLE tags MODIFY name varchar(255) CHARACTER SET utf8 COLLATE utf8_bin;")``` command which is not present in this migration, Yes but this is in gem ```acts_as_taggable```.

Comment (vishalzambre) [inline core/db/migrate/20160511072249_change_collation_for_tag_names.rb]: @krzysiek1507 I'm not getting why it is executing migrations from gem and if it is then it should have run all migration from gem

Comment (krzysiek1507) [inline core/db/migrate/20160511072249_change_collation_for_tag_names.rb]: @vishalzambre Can you check how customize this gem?

Comment (vishalzambre) [inline core/db/migrate/20160511072249_change_collation_for_tag_names.rb]: @krzysiek1507  Yes sure I'll check

**Pull request merged**